drivers/setup/blue/blue.c

```diff
@@ -1206,10 +1450,23 @@ ScrIoControl(
             PUCHAR Src, Dest;
             UINT32 SrcDelta, DestDelta, i;
 
+            /* Validate output buffer */
+            if (stk->Parameters.DeviceIoControl.OutputBufferLength < sizeof(CONSOLE_DRAW))
+            {
+                Status = STATUS_INVALID_PARAMETER;
+                break;
+            }
+            ASSERT(Irp->MdlAddress);
+            ConsoleDraw = (PCONSOLE_DRAW)MmGetSystemAddressForMdl(Irp->MdlAddress);
+            if (ConsoleDraw == NULL)
+            {
+                Status = STATUS_INSUFFICIENT_RESOURCES;
+                break;
+            }
+
             if (DeviceExtension->Enabled && DeviceExtension->VideoMemory)
             {
-                ConsoleDraw = (PCONSOLE_DRAW) MmGetSystemAddressForMdl(Irp->MdlAddress);
-                Src = (PUCHAR) (ConsoleDraw + 1);
+                Src = (PUCHAR)(ConsoleDraw + 1);
                 SrcDelta = ConsoleDraw->SizeX * 2;
```

Comment:
Need to verify that `SrcDelta * ConsoleDraw->SizeY` fits in `OutputBufferLength` in addition to the `CONSOLE_DRAW` structure.